dbt_subprojects/dex/models/_log_decoded_trades/arbitrum/decoded-in-flight/uniswap-v2/uniswap_v2_forks_base_trades_arbitrum.sql

```diff
@@ -0,0 +1,40 @@
+{{ config(
+
+        schema = 'dex_mass_decoding_arbitrum',
+        alias = 'uniswap_v2_base_trades',
+        partition_by = ['block_month'],
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['tx_hash', 'evt_index'],
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+        )
+}}
+
+WITH all_decoded_trades AS (
+    {{
+        uniswap_v2_forks_trades(
+            blockchain = 'arbitrum'
+            , version = 'null'
+            , project = 'null'
+            , Pair_evt_Swap = ref('uniswap_v2_pool_decoding_arbitrum')
+            , Factory_evt_PairCreated = ref('uniswap_v2_factory_decoding_arbitrum')
+        )
+    }}
+)
+
+SELECT uniswap_v2_base_trades.*
+FROM all_decoded_trades AS uniswap_v2_base_trades
+INNER JOIN (
+    SELECT
+        count(*) as transfer_count,
+        contract_address,
+        tx_hash
+    FROM {{ source('tokens', 'transfers') }}
+    WHERE blockchain = 'arbitrum'
+    GROUP BY contract_address, tx_hash
+    HAVING count(*) >= 1
+) AS transfers
+ON (transfers.tx_hash = uniswap_v2_base_trades.tx_hash)
+   AND (transfers.contract_address = uniswap_v2_base_trades.token_bought_address
+        OR transfers.contract_address = uniswap_v2_base_trades.token_sold_address)
```

Comment:
This approach may still fail with oneinch AR trades:
https://dune.com/queries/4103001

Do you think we should recover to use the `excluding oneinch.swaps` approach?

Reply:
hmm so when you query `tokens.transfers` for these troubled tx's, you do in fact find the `tx_hash` values there and sending from one address to another? and the addresses there show weth <--> SF?

Reply:
I also can try double filters.

Reply:
oh my bad, of course this filter is not gonna work if we only look for one token being present. 

We need both `token_sold` and `token_bought` to show up in the transfer table, atm we only check if one of them shows up.

```
ON (transfers.tx_hash = uniswap_v2_base_trades.tx_hash)
   AND (transfers.contract_address = uniswap_v2_base_trades.token_bought_address
           AND transfers.contract_address = uniswap_v2_base_trades.token_sold_address)
 ```

Reply:
I think I may use `array_agg()` for `contract_address` here.
i.e.
```
SELECT uniswap_v2_base_trades.*
FROM all_decoded_trades AS uniswap_v2_base_trades
INNER JOIN (
    SELECT
        tx_hash,
        array_agg(DISTINCT contract_address) as contract_addresses
    FROM {{ source('tokens', 'transfers') }}
    WHERE blockchain = 'ethereum'
    GROUP BY tx_hash
) AS transfers
ON transfers.tx_hash = uniswap_v2_base_trades.tx_hash
   AND (
       contains(transfers.contract_addresses, uniswap_v2_base_trades.token_bought_address)
       AND contains(transfers.contract_addresses, uniswap_v2_base_trades.token_sold_address)
   )
```

Reply:
not sure why you'd prefer that, do you think it's going to be faster than just doing the above or am I not seeing an issue in my logic?

either way is fine with me, just intuitively feels like this might be slower

Reply:
The problem is `transfers` table use `2` lines to record `bought` and `sold` addresses/transfers. This may fail in INNER JOIN without aggregation or `group by`.